### ISSUE TITLE
Incorrect routing of log entries.

### ISSUE THREAD

**Issue by pr4v33n** [googleapis/nodejs-logging#909]
#### Environment details

  - OS: Google, Container-Optimized OS, 81-12871.148.0 stable
  - Node.js version: 12.16.1
  - npm version: 6.13.4
  - `@google-cloud/logging` version: 8.0.6

#### Steps to reproduce

We have the following log instances.
```
const logging = new Logging({ projectId });
export const fooLog = logging.log('foo-log');
export const barLog = logging.log('bar-log');
export const hamLog = logging.log('ham-log');

export async function logMessage(log: Log, severity: string, data: any): Promise<any> {
  return log.write(
      log.entry(
          {
            severity,
            resource: {
              type: 'gce_instance'
            }
          },
          data
      )
  )
}
```
Later in application code, we have calls writing to those logs.

```
logMessage(fooLog, 'INFO', {message: 'Foo message'});
logMessage(barLog, 'INFO', {message: 'Bar message'});
logMessage(hamLog, 'INFO', {message: 'Ham message'});
```

In Stackdriver log viewer, we noticed "Ham message" appearing in `bar-log`.
This behavior started ever since we upgraded our lib from `7.3.0` to `8.0.6`

Thanks!

**Comment by bcoe** [comment]
@pr4v33n thank you for the bug report, we are starting to dig into this.

**Comment by alexander-fenster** [comment]
Hi @pr4v33n,

I fixed the bug in google-gax and released v2.9.1. Please reinstall your dependencies, make sure that `npm ls google-gax` shows 2.9.1, and check if it helped.

Thank you for the perfect bugreport!
